### PR TITLE
Update specs for Dart Sass's new error format

### PR DESCRIPTION
**Pull request by nex3**
[skip dart-sass]

### CODE REVIEW

**Comment by nshahan** [comment]
There are some warning files and empty output.css files that are deleted by this change. Are those expected?

**Comment by nex3** [comment]
> There are some warning files and empty output.css files that are deleted by this change. Are those expected?

Nope! Fixed.

File: spec/libsass-closed-issues/issue_83/error-dart-sass [hunk 1]
```diff
@@ -1,5 +1,10 @@
...
-  @include colors {
-  ^^^^^^^^^^^^^^^^^^^
-  spec/libsass-issues/issue_83/input.scss 7:3  colors()
-  spec/libsass-issues/issue_83/input.scss 7:3  root stylesheet
+  ,
+7 | /   @include colors {
+8 | |     color: $color;
+9 | |   }
+10| |   border-width: 10px;
+  | '--^
```
Comment (nshahan) [inline spec/libsass-closed-issues/issue_83/error-dart-sass]: It looks like the error indicator "^" moved to a more confusing location. Should it be before the line 
`border-width: 10px;`

File: spec/css/moz_document/functions/interpolated/warning [hunk 2]
```diff
@@ -2,68 +2,119 @@ DEPRECATION WARNING on line 1, column 1 of /sass/spec/css/moz_document/functions
...
 relase. For details, see http://bit.ly/moz-document.
 
-@-moz-document url(#{"sass-lang.com"}) {
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+  ,
+1 | / @-moz-document url(#{"sass-lang.com"}) {
+2 | |   a {type: unquoted full url}
+3 | | }
+4 | | @-moz-document url(#{sa + ss}-lang.com) {
+  | '^
```
Comment (nshahan) [inline spec/css/moz_document/functions/interpolated/warning]: All the indicators in this file moved locations. Which version is more accurate?

File: spec/libsass-closed-issues/issue_83/error-dart-sass (hunk 1 above)
Comment (nshahan) [reply spec/libsass-closed-issues/issue_83/error-dart-sass]: Great! SG.

File: spec/libsass-closed-issues/issue_2307/error-dart-sass [hunk 3]
```diff
@@ -1,6 +1,6 @@
 Error: expected "{".
   ,
-3 | 
+3 | {
   | ^
   '
```
Comment (nshahan) [inline spec/libsass-closed-issues/issue_2307/error-dart-sass]: This error doesn't make sense anymore. Expected "{" and it's pointing to a like that has one.

File: spec/css/moz_document/functions/interpolated/warning [hunk 4]
```diff
@@ -5,9 +5,7 @@ relase. For details, see http://bit.ly/moz-document.
   ,
 1 | / @-moz-document url(#{"sass-lang.com"}) {
 2 | |   a {type: unquoted full url}
-3 | | }
-4 | | @-moz-document url(#{sa + ss}-lang.com) {
-  | '^
+3 | \ }
   '
```
Comment (nshahan) [inline spec/css/moz_document/functions/interpolated/warning]: Are you expecting to no longer have a carrot pointing to the error position for all of these?

File: spec/libsass-closed-issues/issue_2307/error-dart-sass (hunk 3 above)
Comment (nex3) [reply spec/libsass-closed-issues/issue_2307/error-dart-sass]: Oh, good catch. This is another `source_span` bug: https://github.com/dart-lang/source_span/pull/30

File: spec/css/moz_document/functions/interpolated/warning (hunk 4 above)
Comment (nex3) [reply spec/css/moz_document/functions/interpolated/warning]: Yep. It now indicates that the entire first `@-moz-document`, and only that one, is highlighted.

File: spec/libsass-closed-issues/issue_2307/error-dart-sass (hunk 3 above)
Comment (nshahan) [reply spec/libsass-closed-issues/issue_2307/error-dart-sass]: I'm not sure this is an improvement. Pointing to the position after the "{" makes it seem like the parser is expecting to see "{{" but would that actually fix the error? Should the error be a missing "}" instead?

File: spec/libsass-closed-issues/issue_1484/error-dart-sass [hunk 5]
```diff
@@ -1,6 +1,6 @@
 Error: expected "{".
```
Comment (nshahan) [inline spec/libsass-closed-issues/issue_1484/error-dart-sass]: The non dart-sass version of the error here seems to be more informative. Should this be expected "}" instead?

Comment (nex3) [reply spec/libsass-closed-issues/issue_1484/error-dart-sass]: Yes, probably; this was a pre-existing issue, so I've filed a separate bug to track it: https://github.com/sass/dart-sass/issues/577

**Pull request merged**